File: AimPlugin3.0.4/AIM.Annotation/Template/AnatomicEntity.cs
#region License

//L
// 2007 - 2013 Copyright Northwestern University
//
// Distributed under the OSI-approved BSD 3-Clause License.
// See http://ncip.github.com/annotation-and-image-markup/LICENSE.txt for details.
//L

#endregion

using System.Xml.Serialization;
using System.Collections.Generic;

namespace AIM.Annotation.Template
{
	public class AnatomicEntity
	{
		private bool _annotatorConfidence;
		private List<AnatomicEntityCharacteristic> _anatomicEntityCharacteristic = new List<AnatomicEntityCharacteristic>();

		[XmlAttribute("annotatorConfidence")]
		public bool AnnotatorConfidence
		{
			get { return _annotatorConfidence; }
			set { _annotatorConfidence = value; }
		}

		[XmlElement("AnatomicEntityCharacteristic", Namespace = TemplateContainer.Namespace)]
		public List<AnatomicEntityCharacteristic> AnatomicEntityCharacteristic
		{
			get { return _anatomicEntityCharacteristic; }
			set { _anatomicEntityCharacteristic = value; }
		}
	}
}

File: ATB_1.0_src/src/main/java/edu/stanford/isis/atb/domain/template/AnatomicEntity.java
/*L
*  Copyright Northwestern University
*  Copyright Stanford University (ATB 1.0 and ATS 1.0)
*
*  Distributed under the OSI-approved BSD 3-Clause License.
*  See http://ncip.github.com/annotation-and-image-markup/LICENSE.txt for details.
*/



package edu.stanford.isis.atb.domain.template;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.List;

import javax.validation.Valid;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;
import org.simpleframework.xml.core.Commit;

/**
 * @author Vitaliy Semeshko
 */

@Root(name="AnatomicEntity", strict=false)
public class AnatomicEntity extends AbstractRemovableElement {

	@Attribute(required=false)
	private boolean annotatorConfidence;

	@Valid
	@ElementList(name="AnatomicEntityCharacteristic", inline=true, required=false)
	private List<AnatomicEntityCharacteristic> characteristics;
	
	@SuppressWarnings("unused")
	@Commit
	private void afterPropertiesSet() {
		if (characteristics != null) {
			for (AnatomicEntityCharacteristic characteristic : characteristics) {
				characteristic.initRemoveCommand(characteristics);
				characteristic.initContext(characteristics);
			}
		}
	}
	
	public boolean isAnnotatorConfidence() {
		return annotatorConfidence;
	}

	public void setAnnotatorConfidence(boolean annotatorConfidence) {
		this.annotatorConfidence = annotatorConfidence;
	}

	public Collection<AnatomicEntityCharacteristic> getCharacteristics() {
		if (characteristics == null) {
			return Collections.emptyList();
		}
		return characteristics;
	}

	public void addCharacteristic(AnatomicEntityCharacteristic characteristic) {
		if (characteristics == null) {
			characteristics = new ArrayList<AnatomicEntityCharacteristic>();
		}
		characteristics.add(characteristic);
		characteristic.initRemoveCommand(characteristics);
		characteristic.initContext(characteristics);
	}

	@Override
	public String toString() {
		return "Anatomic Entity";
	}
	
}
